darwin/future/tests/core/datasets/fixtures.py

```diff
@@ -0,0 +1,49 @@
+from unittest.mock import MagicMock
+
+from pydantic import BaseModel, ValidationError
+from pytest import fixture
+from requests import HTTPError
+
+from darwin.future.core.client import Client
+
+
+@fixture
+def sad_http_client() -> Client:
+    mock = MagicMock(Client)
+    mock.post.side_effect = HTTPError("error")
+    mock.get.side_effect = HTTPError("error")
+
+    return mock
+
+
+@fixture
+def happy_post_client() -> Client:
+    mock_client = MagicMock(Client)
+    mock_client.post.return_value = {
+        "name": "test-dataset",
+        "slug": "1337",
+        "id": 1,
+        "releases": [],
+    }
+
+    return mock_client
+
+
+@fixture
+def happy_get_client() -> Client:
+    mock_client = MagicMock(Client)
+    mock_client.get.return_value = [
```

Comment:
I've been using the responses module for this, maybe unify our approaches either way?

Reply:
Yeah, sure, got a good link to look at?

Reply:
https://github.com/v7labs/darwin-py/blob/090da8d105e4fdb4ab4411510d9448d243c48fb2/darwin/future/tests/core/test_client.py#L59

Can mock the return value and endpoints. There's a bunch of other assertions it gives access to as well but I haven't used too many yet.

Reply:
But the `Client` doesn't return a `Response` object does it?  I thought that was returning `JSONType`?

Reply:
yeah it responds with `JSONType` but the response module you can tell it what an endpoint returns

so in the test below that uses it, might look something like 
```python
with responses.RequestsMock() as rsps:
       endpoint = base_client.config.api_endpoint + "datasets"
       with responses.RequestsMock() as rsps:
        rsps.add(responses.POST, endpoint, json=mock_json_result, status=200)
        dataset = create_dataset(client, mock_json_result["name"])
        assert dataset.name == mock_json_result["name"]
```

Reply:
Ah I'll take a look, cheers